server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java

```diff
@@ -754,4 +759,31 @@ private static IndexMetaData newIndexMeta(String name, Settings oldIndexSettings
             .build();
         return IndexMetaData.builder(name).settings(build).build();
     }
+
+    public void testSerializationRandomVersion() throws IOException {
+        SimpleQueryStringBuilder queryBuilder = new SimpleQueryStringBuilder("query").field("field");
+        Version version = VersionUtils.randomVersion(random());
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            out.setVersion(version);
+            queryBuilder.writeTo(out);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(out.bytes().streamInput(), namedWriteableRegistry())) {
+                in.setVersion(version);
+                SimpleQueryStringBuilder deserializedQuery = new SimpleQueryStringBuilder(in);
+                assertNotSame(queryBuilder, deserializedQuery);
+                assertEquals("query", deserializedQuery.value());
+                assertEquals(1, deserializedQuery.fields().size());
+                assertEquals("field", deserializedQuery.fields().keySet().iterator().next());
+            }
+        }
+    }
+
+    public void testReadFrom_5_6() throws IOException {
```

Comment:
We should add a simple rest test for the `simple_query_string` as a follow up in master. This would have caught this bug if we had one.

Reply:
right, this unit test would have failed if we had it, but it only catches the bug when reading on 6.7 something that was written on 5.6. The other way around is much harder to test in 6.x cause we don't really have the reading code from 5.6.